lib/src/execute.rs

```diff
@@ -403,6 +421,22 @@ impl ExecuteCtx {
         // Invoke the entrypoint function and collect its exit code
         let result = main_func.call_async(&mut store, ()).await;
 
+        // If we collected a profile, write it to the file
+        if let (Some(profile), Some(path)) =
+            (store.data_mut().take_guest_profiler(), guest_profile_path)
+        {
+            if let Err(e) = std::fs::File::create(&path)
+                .map_err(anyhow::Error::new)
+                .and_then(|output| profile.finish(std::io::BufWriter::new(output)))
+            {
+                eprintln!("failed writing profile at {}: {e:#}", path.display());
+            } else {
+                eprintln!();
+                eprintln!("Profile written to: {}", path.display());
+                eprintln!("View this profile at https://profiler.firefox.com/.");
```

Comment:
**Newbie question:** why are we writing what looks to be a success message to stderr?

Reply:
I had done it that way in the equivalent code in Wasmtime to reduce the chance that this message might get mixed up with regular output from the wasm guest. I don't feel strongly about whether Viceroy should do the same, where guest output is less likely to be significant.